build.gradle

```diff
@@ -2,33 +2,32 @@ buildscript {
   ext.deps = [
       'androidTools': [
           'dalvikDx': 'com.jakewharton.android.repackaged:dalvik-dx:9.0.0_r3',
-          'binaryResources': 'com.android.tools.apkparser:binary-resources:27.2.0-alpha08',
+          'binaryResources': 'com.android.tools.apkparser:binary-resources:27.2.2',
           'apkSigner': 'com.android.tools.build:apksig:4.2.2',
-          'bundleTool': 'com.android.tools.build:bundletool:1.2.0',
+          'bundleTool': 'com.android.tools.build:bundletool:1.6.1',
           // Keep this value in sync with bundletool's dependencies.
-          'aapt2Proto': 'com.android.tools.build:aapt2-proto:4.1.0-alpha01-6193524',
+          'aapt2Proto': 'com.android.tools.build:aapt2-proto:4.2.2-7147631',
       ],
-      'clikt': 'com.github.ajalt.clikt:clikt:3.0.0-rc',
-      'junit': 'junit:junit:4.13',
-      'truth': 'com.google.truth:truth:1.0.1',
-      'guava': 'com.google.guava:guava:28.1-jre',
-      'okio': 'com.squareup.okio:okio:2.8.0',
+      'clikt': 'com.github.ajalt.clikt:clikt:3.5.0',
+      'junit': 'junit:junit:4.13.2',
+      'truth': 'com.google.truth:truth:1.1.3',
+      'guava': 'com.google.guava:guava:31.1-jre',
+      'okio': 'com.squareup.okio:okio:3.2.0',
       'byteunits': 'com.jakewharton.byteunits:byteunits:0.9.1',
-      'asm': 'org.ow2.asm:asm:9.0-beta',
-      'diffUtils': 'io.github.java-diff-utils:java-diff-utils:4.7',
-      'picnic': 'com.jakewharton.picnic:picnic:0.5.0',
+      'asm': 'org.ow2.asm:asm:9.3',
+      'diffUtils': 'io.github.java-diff-utils:java-diff-utils:4.12',
+      'picnic': 'com.jakewharton.picnic:picnic:0.6.0',
   ]
 
   dependencies {
-    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.0"
-    classpath 'org.jetbrains.dokka:dokka-gradle-plugin:0.10.0'
-    classpath 'org.jlleitschuh.gradle:ktlint-gradle:9.3.0'
-    classpath 'com.vanniktech:gradle-maven-publish-plugin:0.9.0'
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.10"
+    classpath 'org.jetbrains.dokka:dokka-gradle-plugin:1.7.10'
+    classpath 'org.jlleitschuh.gradle:ktlint-gradle:11.0.0'
+    classpath 'com.vanniktech:gradle-maven-publish-plugin:0.22.0'
```

Comment:
Not sure this one is entirely safe to do. I remember a bunch of migrations needed in our other open source projects.

Reply:
Migrated Kotlin & Ktlint, Dokka & Publish seems not to be used,

Reply:
They're used in the release process